src/core/tests/type_prop/broadcast.cpp

```diff
@@ -915,4 +919,308 @@ TEST(type_prop, broadcast_i32_shape_value) {
     broadcast_v3->revalidate_and_infer_types();
 
     ASSERT_EQ(broadcast_v3->get_output_partial_shape(0), PartialShape({5, -1}));
-}
\ No newline at end of file
+}
+
+TEST(type_prop, broadcast_v3_default_constructor) {
+    auto param = make_shared<op::Parameter>(element::f32, Shape{5, 2, 3, 1});
+    auto target_shape = op::Constant::create<int64_t>(element::i64, Shape{3}, {1, 3, 6});
+
+    auto op = make_shared<op::v3::Broadcast>();
+
+    EXPECT_EQ(op->get_broadcast_spec().m_type, BroadcastType::NUMPY);
+
+    op->set_broadcast_spec(BroadcastType::BIDIRECTIONAL);
+    EXPECT_EQ(op->get_broadcast_spec().m_type, BroadcastType::BIDIRECTIONAL);
+
+    op->set_argument(0, param);
+    op->set_argument(1, target_shape);
+    op->validate_and_infer_types();
+
+    EXPECT_EQ(op->get_element_type(), element::f32);
+    EXPECT_EQ(op->get_shape(), (Shape{5, 2, 3, 6}));
+}
+
+TEST(type_prop, broadcast_v3_bidirectional_data_bigger_rank_numpy) {
+    auto param = make_shared<op::Parameter>(element::f32, Shape{5, 2, 3, 1});
+    auto target_shape = op::Constant::create<int64_t>(element::i64, Shape{3}, {4, 3, 6});
+
+    OV_EXPECT_THROW(make_shared<op::v3::Broadcast>(param, target_shape),
+                    NodeValidationFailure,
+                    HasSubstr("Broadcast target_shape has smaller rank"));
+}
+
+// Interval dimensions with labels
+TEST(type_prop, broadcast_v3_labels_in0_dynamic_mixed_dims_bidirectional) {
+    // All dimensions of A have labels, B without labels
+    PartialShape pshape_A{-1, 2, 1, {4, 8}, -1, {4, 8}, -1, {1, 8}, {1, 10}, {4, 18}};
+    PartialShape pshape_B{-1, 2, {3, 9}, 1, {3, 9}, -1, {1, 9}, -1, {3, 19}, {1, 10}};
+
+    PartialShape expected_shape = {-1, 2, {3, 9}, {4, 8}, {3, 9}, {4, 8}, -1, -1, {3, 19}, {4, 18}};
+    std::vector<size_t> expected_labels{10, 11, 0, 13, 0, 15, 16, 17, 0, 19};
+
+    set_shape_labels(pshape_A, {10, 11, 12, 13, 14, 15, 16, 17, 18, 19});
+    set_shape_labels(expected_shape, expected_labels);
+
+    auto data = std::make_shared<op::Parameter>(element::f32, pshape_A);
+    auto target_shape = std::make_shared<op::Parameter>(element::f32, pshape_B);
+    auto shape_of = make_shared<op::v3::ShapeOf>(target_shape);
+
+    auto op = make_shared<op::v3::Broadcast>(data, shape_of, "BIDIRECTIONAL");
+
+    const auto out_shape = op->get_output_partial_shape(0);
+
+    EXPECT_EQ(out_shape, expected_shape);
```

Comment:
Check if `expected_shape` is required in test I think in this case can be removed and shape can specified in this assertion.
Set labels on expected shape is also not required.